packages/wrangler/src/versions/list.ts

```diff
@@ -1,93 +1,79 @@
-import path from "path";
 import { logRaw } from "@cloudflare/cli";
-import { findWranglerConfig, readConfig } from "../config";
+import { readConfig } from "../config";
+import { createCommand } from "../core/create-command";
 import { UserError } from "../errors";
 import * as metrics from "../metrics";
-import { printWranglerBanner } from "../update-check";
 import { requireAuth } from "../user";
 import formatLabelledValues from "../utils/render-labelled-values";
 import { fetchDeployableVersions } from "./api";
-import type {
-	CommonYargsArgv,
-	StrictYargsOptionsToInterface,
-} from "../yargs-types";
 import type { ApiVersion, VersionCache } from "./types";
 
 const BLANK_INPUT = "-"; // To be used where optional user-input is displayed and the value is nullish
 
-export type VersionsListArgs = StrictYargsOptionsToInterface<
-	typeof versionsListOptions
->;
-
-export function versionsListOptions(yargs: CommonYargsArgv) {
-	return yargs
-		.option("name", {
-			describe: "Name of the worker",
+export const versionsListCommand = createCommand({
+	metadata: {
+		description: "List the 10 most recent Versions of your Worker",
+		owner: "Workers: Authoring and Testing",
+		status: "open-beta",
+	},
+	args: {
+		name: {
+			describe: "Name of the Worker",
 			type: "string",
 			requiresArg: true,
-		})
-		.option("json", {
+		},
+		json: {
 			describe: "Display output as clean JSON",
 			type: "boolean",
 			default: false,
-		});
-}
+		},
+	},
+	handler: async function versionsSecretListHandler(args) {
+		const config = readConfig(args.config, args, false, true);
```

Comment:
Do we need this?

Reply:
Updated here: https://github.com/cloudflare/workers-sdk/pull/7437/commits/25c9e043febc6597c04036501909c771b2632722